From: Andrea Azzarone <andrea.azzarone@canonical.com>
Date: Fri, 3 Aug 2018 15:43:01 +0300
Subject: [PATCH 23/25] snap: Don't treat auth cancellation as an error

SNAPD_ERROR_PERMISSION_DENIED is triggered when the user cancels the PolicyKit
authorization process. For this reasons treat this error in a special way to
avoid showing an error notification.

Fixes: https://bugs.launchpad.net/bugs/1785240
---
 meson.build                   | 2 +-
 plugins/snap/gs-plugin-snap.c | 3 +++
 2 files changed, 4 insertions(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index f5a80ce..5471426 100644
--- a/meson.build
+++ b/meson.build
@@ -171,7 +171,7 @@ if get_option('gudev')
 endif
 
 if get_option('snap')
-  snap = dependency('snapd-glib', version : '>= 1.42')
+  snap = dependency('snapd-glib', version : '>= 1.43')
 endif
 
 gnome = import('gnome')
diff --git a/plugins/snap/gs-plugin-snap.c b/plugins/snap/gs-plugin-snap.c
index 5507abc..78d4646 100644
--- a/plugins/snap/gs-plugin-snap.c
+++ b/plugins/snap/gs-plugin-snap.c
@@ -145,6 +145,9 @@ snapd_error_convert (GError **perror)
 		case SNAPD_ERROR_TWO_FACTOR_INVALID:
 			error->code = GS_PLUGIN_ERROR_AUTH_INVALID;
 			break;
+		case SNAPD_ERROR_AUTH_CANCELLED:
+			error->code = GS_PLUGIN_ERROR_CANCELLED;
+			break;
 		case SNAPD_ERROR_CONNECTION_FAILED:
 		case SNAPD_ERROR_WRITE_FAILED:
 		case SNAPD_ERROR_READ_FAILED:
